### PR TITLE
Issue #6207: Added Xpath IT Regression test for ParameterAssignmentCheck

### PR DESCRIPTION
**Pull request by MANISH-K-07**
Part of #6207 

Added new Xpath IT Regression test for ParameterAssignmentCheck
The check doesn't involve usage of config properties. So, created two files for different scenarios

`Update` : Created one more file to show violations in _**Ctor**_ parameter assignment.

### CODE REVIEW

**Comment by MANISH-K-07** [comment]
@romani , Is this okay?

**Review by romani — CHANGES_REQUESTED** [review]
We requesting 3 test method.
 Please add test for assignments in ctor

File: src/it/resources/org/checkstyle/suppressionxpathfilter/parameterassignment/SuppressionXpathRegressionParameterAssignmentLambdas.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.checkstyle.suppressionxpathfilter.parameterassignment;
+
+public class SuppressionXpathRegressionParameterAssignmentLambdas {
+
+     public interface SomeInterface {
+        void method(int a);
+    }
+
```
Comment (MANISH-K-07) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/parameterassignment/SuppressionXpathRegressionParameterAssignmentLambdas.java:8]: @romani , I had a doubt regarding Checkstyle's indentation checks.

In this file, the indentation of line 5 is generally matched with line 7.
I unknowingly added the extra whitespace character before "public" in line 5 but the checks have all passed.

I wanted to clarify with you if this is expected or not ? Should checkstyle be reporting this as violation or not ?

**Comment by MANISH-K-07** [comment]
@romani , One more thing that I wanted to bring to your notice.

The Java input files related to ParameterAssignmentCheck of our src/test directory have no coverage of violations in Ctors. 

Should I expand the coverage of the test  and introduce new violations to the input files to show the usage of check in constructors ??

- [InputParameterAssignmentReceiver.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentReceiver.java)

- [InputParameterAssignmentWithUnchecked.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentWithUnchecked.java)

File: src/it/resources/org/checkstyle/suppressionxpathfilter/parameterassignment/SuppressionXpathRegressionParameterAssignmentLambdas.java (hunk 1 above)
Comment (nrmancuso) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/parameterassignment/SuppressionXpathRegressionParameterAssignmentLambdas.java:8]: @MANISH-K-07 our checks for test inputs are more liberal than our checks for production code. Think about it like this: we need to have weird/ugly input for checks to have something to complain about in tests :)

**Review by nrmancuso — APPROVED** [review]

**Comment by MANISH-K-07** [comment]
I have added test for Ctor assignments here in response to https://github.com/checkstyle/checkstyle/pull/14589#pullrequestreview-1914478089.

> @romani , One more thing that I wanted to bring to your notice.
> 
> The Java input files related to ParameterAssignmentCheck of our src/test directory have no coverage of violations in Ctors.
> 
> Should I expand the coverage of the test and introduce new violations to the input files to show the usage of check in constructors in another PR ??
> 
> * [InputParameterAssignmentReceiver.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentReceiver.java)
> * [InputParameterAssignmentWithUnchecked.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentWithUnchecked.java)

@nrmancuso ,
Please share your opinion on this.

**Comment by MANISH-K-07** [comment]
I have added test for Ctor assignments here in response to https://github.com/checkstyle/checkstyle/pull/14589#pullrequestreview-1914478089.

> @romani , One more thing that I wanted to bring to your notice.
> 
> The Java input files related to ParameterAssignmentCheck of our src/test directory have no coverage of violations in Ctors.
> 
> Should I expand the coverage of the test and introduce new violations to the input files to show the usage of check in constructors in another PR ??
> 
> * [InputParameterAssignmentReceiver.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentReceiver.java)
> * [InputParameterAssignmentWithUnchecked.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentWithUnchecked.java)

@romani , @nrmancuso ,
Please share your opinion on this.

**Pull request merged**